src/internal/connector/onedrive/collections_test.go

```diff
@@ -1431,3 +1432,94 @@ func delItem(
 
 	return item
 }
+
+func (suite *OneDriveCollectionsSuite) TestCollectItems() {
+	next := "next"
+	delta := "delta"
+
+	syncStateNotFound := "SyncStateNotFound" // TODO(meain): export graph.errCodeSyncStateNotFound
+	me := odataerrors.NewMainError()
+	me.SetCode(&syncStateNotFound)
+
+	deltaError := odataerrors.NewODataError()
+	deltaError.SetError(me)
+
+	table := []struct {
+		name             string
+		items            []deltaPagerResult
+		deltaURL         string
+		prevDelta        string
+		prevDeltaSuccess bool
+		err              error
+	}{
+		{
+			name:     "delta on first run",
+			deltaURL: delta,
+			items: []deltaPagerResult{
+				{deltaLink: &delta},
+			},
+			prevDeltaSuccess: true,
+		},
+		{
+			name:     "next then delta",
+			deltaURL: delta,
+			items: []deltaPagerResult{
+				{nextLink: &next},
+				{deltaLink: &delta},
+			},
+			prevDeltaSuccess: true,
+		},
+		{
+			name:     "invalid prev delta",
+			deltaURL: delta,
+			items: []deltaPagerResult{
+				{nextLink: &next, err: deltaError},
+				{deltaLink: &delta}, // works on retry
```

Comment:
nit: could you use a different value here besides `next` and `delta` please just so we can be certain we're returning what we think we are? It's being a bit paranoid, but it also helps reduce the number of assumptions about how the code could be broken 😅

Reply:
I'm not sure I understand you.
*Related: The next is useless there and I have removed it*

Reply:
oh, I actually totally misread this bit of code. You can ignore this comment 😅